internal/broker/config.go

```diff
@@ -31,7 +34,27 @@ const (
 func parseConfigFile(cfgPath string) (userConfig, error) {
 	cfg := userConfig{}
 
-	iniCfg, err := ini.Load(cfgPath)
+	// Check if a .d directory exists and add all files in it to the sources.
+	dropInDir := cfgPath + ".d"
+	exists, err := fileutils.FileExists(dropInDir)
+	if err != nil {
+		return cfg, err
+	}
+	var dropInFiles []interface{}
+	if exists {
+		files, err := os.ReadDir(dropInDir)
+		if err != nil {
+			return cfg, err
+		}
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+			dropInFiles = append(dropInFiles, filepath.Join(dropInDir, file.Name()))
+		}
+	}
```

Comment:
```suggestion
	files, err := os.ReadDir(dropInDir)
	if err != nil && !errors.Is(fs.ErrNotExist) {
	       return cfg, nil
	}
	var dropInFiles any
	for _, file := range files {
		if file.IsDir() {
			continue
		}
		dropInFiles = append(dropInFiles, filepath.Join(dropInDir, file.Name()))
	}
```

Otherwise instead of `fileutils.FileExists` can just use `os.Stat` and check also for being a dir.

Reply:
> files, err := os.ReadDir(dropInDir)
	if err != nil && !errors.Is(fs.ErrNotExist) {
	       return cfg, nil
	}

That has the effect that we ignore e.g. permission errors. The purpose of 

```
exists, err := fileutils.FileExists(dropInDir)
	if err != nil {
		return cfg, err
	}
```

is to fail in such a case, for example when the directory exists but can't be accessed.

Reply:
Sure, but wouldn't `os.ReadDir(dropInDir)` fail anyways in such case?

Reply:
I think we don't understand each other. Maybe we can solve this better in a quick call?